src/App.tsx

```diff
@@ -14,11 +15,144 @@ const goodsFromServer: string[] = [
   'Garlic',
 ];
 
-const App: React.FC = () => (
-  <div className="App">
-    <h1>Goods</h1>
-    {goodsFromServer.length}
-  </div>
-);
+type State = {
+  isStart: boolean;
+  isVisible: boolean;
+  listOfGoods: string[];
+};
+
+class App extends React.Component<{}, State> {
+  state = {
+    isStart: false,
+    isVisible: true,
+    listOfGoods: goodsFromServer,
+  };
+
+  startButtonHandler = (): void => (
+    this.setState({
+      isStart: true,
+      isVisible: false,
+    })
+  );
+
+  reversList = () => (
+    this.setState(state => (
+      {
+        listOfGoods: [...state.listOfGoods].reverse(),
+      }
+    ))
+  );
+
+  alfabetSortList = () => (
+    this.setState(state => (
+      {
+        listOfGoods: [...state.listOfGoods].sort((a, b) => (
+          a.localeCompare(b)
+        )),
+      }
+    ))
+  );
+
+  sortListByLength = () => (
+    this.setState(state => (
+      {
+        listOfGoods: [...state.listOfGoods].sort((a, b) => (
+          a.length - b.length
+        )),
+      }
+    ))
+  );
+
+  resetList = () => (
+    this.setState(
+      {
+        listOfGoods: goodsFromServer,
+      },
+    ));
+
+  render() {
+    const { isStart, isVisible, listOfGoods } = this.state;
+
+    return (
+      <div className="container">
+        <button
+          type="button"
+          className={classNames(
+            'btn',
+            {
+              hidden: isStart,
+            },
+          )}
+          onClick={this.startButtonHandler}
+        >
+          Start
+        </button>
+        <ul className={classNames(
+          'listOfGoods',
+          {
+            hidden: isVisible,
+          },
+        )}
+        >
+          {listOfGoods.map(good => (
+            <li key={good}>
+              {good}
+            </li>
+          ))}
+        </ul>
+        <div className="buttons">
+          <button
+            type="button"
+            className={classNames(
+              'btn',
+              {
+                hidden: isVisible,
+              },
+            )}
+            onClick={this.reversList}
+          >
+            Reverse
+          </button>
+          <button
+            type="button"
+            className={classNames(
+              'btn',
+              {
+                hidden: isVisible,
```

Comment:
```suggestion
                hidden: isVisible,
```
Щоб не прописувати hidden: isVisible для кожної кнопки і дублювати кол, можна прописати це лише раз для контейнера buttons

Reply:
Дякую, врахую на майбутне